include/core/adapter/Adapter2D.hpp

```diff
@@ -0,0 +1,69 @@
+#ifndef GENERIC_ADAPTER_H
+#define GENERIC_ADAPTER_H
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/geometries.hpp>
+#include <HY_Catchment.hpp>
+#include <HY_HydroLocation.hpp>
+
+using namespace hy_features::hydrolocation;
+class GenericAdapter
+{
+    public:
+
+    typedef long time_step_t;
+
+
+    // create type aliases
+    using Models = std::vector<std::string>;
+    using long = time_point_t;
+    using long = time_delta;
+
+    GenericAdapter(std::string adapter_id, Models ls, Models rs);
+
+    virtual ~Adapter2D();
+
+    /** Contribute a flux with id flux_id from source contrib_id at time step t*/
+
+    virtual void add_flux(std::string flux_id, std::string contrib_id, time_point_t t, time_delta t1, double* vals)= 0;
```

Comment:
This isn't necessarily something that needs to be changed, but I am curious about it.

My understanding was that this type is a more general abstraction for something like the (HY_Features-specific) _HY_HydroNexus_ (if that isn't correct, then this comment can probably be ignored).  Should we apply the relationship at this point?   Also, is the use of slightly different terminology (e.g., "flux" here versus "flow" in _HY_HydroNexus_) just due to some dialect differences when talking about similar concepts in the different contexts?